### DISCUSSION TITLE
Visual comparison of output quality between TeX and Typst?

### DISCUSSION THREAD

**Discussion by user202729** [typst/typst#336] [General]
I think this may convince some people to switch.

We can do something similar to this comparison:

![](https://i.stack.imgur.com/OBiLO.png)

source: https://tex.stackexchange.com/a/110140/250119

**Comment by Enivex** [comment]
To be fair, any comparison should also include the case where microtype is loaded. There's very little reason not to use it in LaTeX, as it's basically just a set-and-forget way to make documents look significantly better.

**Comment by ghost** [comment]
The picture counts only the number of hyphenations. What I miss is: are they good?
`abovementioned` should semantically be split as `above-mentioned` and not as `abovemen-tioned` (the word is not used in the image, just an example).
Probably no program exists that can prefer split positions. The current line splitting algorithm used in Typst could do this by including this in the cost model. But it would require good data for all languages.

Also, some fonts have good looking glyphs for the hyphenation symbol. It may have a different effect then.

All of the examples above have the last line shorter than all other lines.
When I look at old handwritten texts by professionals, they would probably not do this.
But I think TeX can achieve that with some command.

Regarding `microtype`: it has some good features but by default it also shrinks and stretches some glyphs, which is not good in my opinion. Of course this can get disabled.

**Comment by alerque** [comment]
I would be interested in curating a repository of typesetter comparisons. I ~~would suggest~~ have initialized this as a Git repository with various inputs (equivalent inputs for different typesetting engines) and build rules that automatically generate outputs. Besides being a comparison of outputs, this would double as a kind of Rosetta stone that showcases of how different typesetters approach things and might help people that are experienced in one but need to do some work in another.

(Side note: as the maintainer of the [SILE typesetter](https://sile-typesetter.org) I have an obvious preference, but I don't see this as a way to only showcase only the things one engine is better at, I believe there are pros and cons of all the major options and showcasing the *best* efforts for each engine would be the most useful for end users).

Earlier today I ran a quick side by side test for #1620. I've now repurposed that bug MWE as a starting point for this project. Sources are in https://github.com/alerque/polytype. I would love others to contribute both to the website aspect and interesting typesetter inputs. So far I have XeLaTeX, Typst, and SILE working to generate PDFs in CI that are posted to a static website (currently at https://alerque.github.io/polytype, soon to be https://polytype.dev).
